app/controllers/teams_controller.rb

```diff
@@ -76,11 +76,31 @@ def delete_membership
     redirect_to @team
   end
 
+  def assign_membership_by_email
+    user = User.find_by_email params[:email]
+    if user
+      assign_membership_to_user user
+    else
+      flash[:error] = I18n.t('teams.invite_user_to_team.no_user_for_mail')
+    end
+    redirect_to @team
+  end
+
   def current_ability
     @current_ability ||= Ability.new(current_user, params[:team_member])
   end
 
   private
+    def assign_membership_to_user (new_member)
```

Comment:
@bakoe For now I've implemented all checks right here, in teams_controller. Not sure if we want to think about moving those checks somewhere (into the model?)

Reply:
I think for the moment this is the right place to put these checks – the model is probably going to get some refactoring regarding the primitive operations on `owners` and `members` anyway (see #203). Maybe this will lead to a model-side check regarding i. e. whether the newly added member is already a member of the team. But as long as this is not handled globally in the model, I think it's a good way to put the checks into the controller as is.